### PR TITLE
Reduce deprecation warnings (refs: #3192)

### PR DESCRIPTION
**Pull request by tk0miya**
Could you review this please?

### CODE REVIEW

**Comment by tk0miya** [comment]
refs: #3192

File: tests/test_ext_graphviz.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def decorator(app, *args, **kwargs):
                 dot = subprocess.Popen([graphviz_dot, '-V'],
                                        stdout=subprocess.PIPE,
                                        stderr=subprocess.PIPE)  # show version
-                dot.wait()
```
Comment (shimizukawa) [inline tests/test_ext_graphviz.py:29]: Why is this changed?

**Review by shimizukawa — APPROVED** [review]
LGTM with nits ;)

File: tests/test_ext_graphviz.py (hunk 1 above)
Comment (tk0miya) [reply tests/test_ext_graphviz.py:29]: It suppresses `ResourceWarning`:
```
/home/travis/build/sphinx-doc/sphinx/tests/util.py:271: ResourceWarning: unclosed file <_io.BufferedReader name=10>
  func(app, status, warning, *args2, **kwargs2)
```
https://travis-ci.org/sphinx-doc/sphinx/jobs/184006704#L1305

It says the pipes for `Popen()` are not closed on `Popen.wait()`. So I use `Popen.communicate()` to close them politely.

Comment (shimizukawa) [reply tests/test_ext_graphviz.py:29]: I see. Thanks for explanation. I didn't know that.
